eztrace: 1.1-11 -> 2.1.1, fix build, adopt, otf2: fix cross

xokdvium opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

Sigmanificient commented on pkgs/by-name/ez/eztrace/package.nix:
```diff
@@ -2,43 +2,77 @@
+  libotf2,
 }:
 
 stdenv.mkDerivation rec {
```
Can you make the derivation use `finalAttrs`?

antonmosich commented on pkgs/by-name/ez/eztrace/package.nix:
```diff
@@ -2,43 +2,81 @@
     description = "Tool that aims at generating automatically execution trace from HPC programs";
-    license = licenses.cecill-b;
-    maintainers = [ ];
+    license = lib.licenses.cecill-b;
```
```suggestion
    homepage = "https://eztrace.gitlab.io/eztrace/index.html";
    license = lib.licenses.cecill-b;
```

antonmosich reviewed (commented):
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 388855`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 6 packages built:</summary>
  <ul>
    <li>eztrace</li>
    <li>eztrace.lib</li>
    <li>eztrace.man</li>
    <li>libotf2</li>
    <li>libotf2.doc</li>
    <li>libotf2.lib</li>
  </ul>
</details>

awwpotato commented:
eztrace is already being updated and cleaned up by #395939

xokdvium committed:
otf2: enableParallelBuilding and enableParallelChecking

xokdvium committed:
otf2: fix cross

xokdvium committed:
otf2: split out and lib

awwpotato commented on pkgs/by-name/ez/eztrace/package.nix:
```diff
@@ -2,43 +2,84 @@
     description = "Tool that aims at generating automatically execution trace from HPC programs";
-    license = licenses.cecill-b;
-    maintainers = [ ];
+    homepage = "https://eztrace.gitlab.io/eztrace/index.html";
```
```suggestion
    homepage = "https://eztrace.gitlab.io/eztrace/index.html";
    downloadPage = "https://gitlab.com/eztrace/eztrace/";
```

awwpotato replied:
not resolved

awwpotato commented:
> fyi it seems that `nix-update-script` doesn't do the right thing for `eztrace`. For me it downgrades from `2.1.1` -> `2.1`. I'm not very familiar with how it works under the hood and I'm not sure how to fix that. Since you enabled that in #395939 maybe you have some ideas?

try 
```nix
nix-update-script { extraArgs = [ "--version-regex=eztrace-(.*)" ]; }
```

awwpotato commented:
> fyi it seems that `nix-update-script` doesn't do the right thing for `eztrace`. For me it downgrades from `2.1.1` -> `2.1`. I'm not very familiar with how it works under the hood and I'm not sure how to fix that. Since you enabled that in #395939 maybe you have some ideas?

It's because nix-update uses the latest release if one is found https://github.com/Mic92/nix-update/blob/d3a58c41034c788e7917ea09fe051acbe56dc5c4/nix_update/version/gitlab.py#L40. Someone should probably submit a feature request for always using the latest tag.

awwpotato commented on pkgs/by-name/ez/eztrace/package.nix:
```diff
@@ -2,43 +2,84 @@
+  postPatch = ''
+    substituteInPlace src/eztrace-lib/eztrace_otf2.c \
+      --replace-fail "/bin/rm" "rm"
+    substituteInPlace cmake_modules/FindOTF2.cmake \
+      --replace-fail "find_program(OTF2_CONFIG otf2-config REQUIRED)" \
+                     "find_program(OTF2_CONFIG "${lib.getExe' otf2 "otf2-config"}" REQUIRED)"
+    substituteInPlace cmake_modules/FindOTF2.cmake \
```
```suggestion
    substituteInPlace cmake_modules/FindOTF2.cmake \
      --replace-fail "find_program(OTF2_CONFIG otf2-config REQUIRED)" \
                     "find_program(OTF2_CONFIG "${lib.getExe' otf2 "otf2-config"}" REQUIRED)" \
```

xokdvium committed:
eztrace: 1.1-11 -> 2.1.1, fix build, adopt

xokdvium committed:
eztrace: add meta.homepage

xokdvium committed:
eztrace: split dev output and fix .pc files

xokdvium committed:
otf2: split doc output

xokdvium committed:
eztrace: add meta.downloadPage

awwpotato commented on pkgs/by-name/ez/eztrace/package.nix:
```diff
@@ -2,43 +2,91 @@
+  postInstall = ''
+    moveToOutput bin/eztrace_create_plugin "$dev"
+  '';
+
```
```suggestion

  doInstallCheck = true;
  nativeInstallCheckInputs = [ versionCheckHook ];

  passthru.updateScript = nix-update-script { };

```

awwpotato replied:
> Sorry, I might be misunderstanding something. What's the purpose of enabling `nix-update-script`, if it doesn't do the right thing here ([#388855 (comment)](https://github.com/NixOS/nixpkgs/pull/388855#issuecomment-2816327760))? 

Oops forgot about that issue.

> Also the same goes for `versionCheckHook`. The version we are packaging `2.1.1` reports as being `2.1.0` (defined [here](https://gitlab.com/eztrace/eztrace/-/blob/2.1.1/CMakeLists.txt?ref_type=tags#L3)), so the hook would have to be disabled either way. I suppose we could patch the sources to report the correct version, but that sort of defeats the purpose of the hook. Do you think that would be worthwhile?

I would patch the version because `versionCheckHook` also ensures that binary is installed correctly and runs, but you don't have to.

xokdvium committed:
eztrace: fix reported version and use versionCheckHook

Upstream seemingly forgot to bump their patch version
for 2.1.1 and do a gitlab release as well.

Use `versionCheckHook` as a smoke test for
installation completeness.

LeSuisse reviewed (approved):
Quick test looks good.


## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 388855`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 7 packages built:</summary>
  <ul>
    <li>eztrace</li>
    <li>eztrace.dev</li>
    <li>eztrace.lib</li>
    <li>eztrace.man</li>
    <li>otf2</li>
    <li>otf2.doc</li>
    <li>otf2.lib</li>
  </ul>
</details>
